[IMP] b_shift: unsubscribe from already generated shifts

remytms opened the pull request:
[task](https://gestion.coopiteasy.be/web#id=1816&view_type=form&model=project.task&action=475&active_id=213)

remytms committed:
[IMP] b_shift: unsubscribe worker from task template

Remove worker from future shift when unsubscribed from task template

remytms committed:
[ADD] b_shift: tests unsubscribe from task_template

remytms committed:
[IMP] b_shift: update already generated shift when changing holiday

remytms committed:
[IMP] b_shift: holiday wizard

Holiday Wizard allows now to modify holiday under certain conditions.

remytms committed:
[IMP] b_shift: (un)subscribe from shift when exemption changes

vvrossem commented:
```
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_change_worker_holiday_1 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Change the holiday start and end time on a cooperative_status. Check that worker is correctly unsbscribed or subscribed again to already generated shifts. 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 196, in test_change_worker_holiday_1 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(self._count_number_of_shift(self.worker_regular_1), 5) 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 6 != 5 
2021-06-16 13:11:40,647 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_change_worker_holiday_2 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Change the holiday start and end time on a cooperative_status. Check that worker is correctly unsbscribed or subscribed again to already generated shifts.  This test uses the wizard. 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 264, in test_change_worker_holiday_2 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(self._count_number_of_shift(self.worker_regular_1), 5) 
2021-06-16 13:11:40,647 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 6 != 5 
2021-06-16 13:11:40,648 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_change_worker_temporary_exemption_1 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Change the temporary exemption start and end time on a cooperative_status. Check that worker is correctly unsbscribed or subscribed again to already generated shifts. 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 342, in test_change_worker_temporary_exemption_1 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(self._count_number_of_shift(self.worker_regular_1), 5) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 6 != 5 
2021-06-16 13:11:40,648 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_change_worker_temporary_exemption_2 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Change the exemption start and end time on a cooperative_status. Check that worker is correctly unsbscribed or subscribed again to already generated shifts.  This test uses the wizard. 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 413, in test_change_worker_temporary_exemption_2 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(self._count_number_of_shift(self.worker_regular_1), 5) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 6 != 5 
2021-06-16 13:11:40,648 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_change_working_mode_1 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Check that changing a regular worker to irregular via the wizard removes the worker from the already generated shifts. 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 170, in test_change_working_mode_1 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(number_shift_worker_1, 2) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 3 != 2 
2021-06-16 13:11:40,648 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_unsubscribe_worker_from_task_template_1 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Check that removing a worker from a task_template via the task_template remove the worker from the already generated shifts. 
2021-06-16 13:11:40,648 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 103, in test_unsubscribe_worker_from_task_template_1 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(number_shift_worker_1, 2) 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 3 != 2 
2021-06-16 13:11:40,649 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ====================================================================== 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAIL: test_unsubscribe_worker_from_task_template_2 (odoo.addons.beesdoo_shift.tests.test_beesdoo_shift.TestBeesdooShift) 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `  Check that removing a worker from a task_template via the res_partner remove the worker from the already generated shifts. 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Traceback (most recent call last): 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `   File "/home/vincent/twelve-odoo/coopiteasy/obeesdoo/beesdoo_shift/tests/test_beesdoo_shift.py", line 137, in test_unsubscribe_worker_from_task_template_2 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: `     self.assertEqual(number_shift_worker_1, 2) 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: ` AssertionError: 3 != 2 
2021-06-16 13:11:40,649 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: Ran 7 tests in 5.340s 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift: FAILED 
2021-06-16 13:11:40,649 47258 INFO twelve-worker_status odoo.addons.beesdoo_shift.tests.test_beesdoo_shift:  (failures=7) 
2021-06-16 13:11:40,649 47258 INFO twelve-worker_status odoo.modules.module: odoo.addons.beesdoo_shift.tests.test_beesdoo_shift tested in 5.34s, 7422 queries 
2021-06-16 13:11:40,649 47258 ERROR twelve-worker_status odoo.modules.module: Module beesdoo_shift: 7 failures, 0 errors 
2021-06-16 13:11:40,679 47258 INFO twelve-worker_status odoo.modules.loading: 16 modules loaded in 7.35s, 7422 queries 
2021-06-16 13:11:40,809 47258 INFO twelve-worker_status odoo.modules.registry: verifying fields for every extended model 
2021-06-16 13:11:41,023 47258 ERROR twelve-worker_status odoo.modules.loading: At least one test failed when loading the modules. 
```

remytms committed:
[CHG] b_shift: show holiday wizard except when resigning

remytms committed:
[IMP] b_shift: subscribe when added to a new task template

This also fix tests and improve unsubscribe method to act like the
subscribe one.

remytms committed:
[FIX] b_shift: subscribe to task template when it's full

remytms committed:
[FIX] b_shift: tests

remytms committed:
[FIX] b_worker_status: tests

remytms committed:
[IMP] b_worker_status: keep alert after a holiday or exempt

When a worker is in alert status and that it take a break (holiday or
exempted) it should stay in alert at the end of his break.
alert_start_time should be updated in consequence.

vvrossem commented on beesdoo_base/views/partner.xml:
```diff
@@ -96,6 +96,16 @@
     </record>
 
+    <record id="view_beesdoo_res_partner_filter" model="ir.ui.view">
+        <field name="field_name">res.partner.select</field>
```
should be `<field name="name">` instead of `<field name="field_name">`

vvrossem commented on beesdoo_shift/demo/templates.xml:
```diff
@@ -60,9 +62,59 @@
+    </record>
+
+    <record id="task_template_2_demo" model="beesdoo.shift.template">
+        <field name="name">A_WEN-14:00-16:00</field>
```
should be `A_WED` instead of `A_WEN`

vvrossem committed:
[FIX] beesdoo_base: correct <field name="name">

vvrossem committed:
[IMP] beesdoo_shift: abbreviation of wednesday is 'WED'
